src/integrators/integrator_utils.jl

```diff
@@ -89,8 +90,9 @@ function savevalues!(integrator::ODEIntegrator,force_save=false,reduce_size=true
         copyat_or_push!(integrator.sol.alg_choice,integrator.saveiter,integrator.cache.current)
       end
     end
+    saved = true
   end
-  if force_save || (integrator.opts.save_everystep && integrator.iter%integrator.opts.timeseries_steps==0)
+  if (force_save && !saved) || (integrator.opts.save_everystep && integrator.iter%integrator.opts.timeseries_steps==0)
```

Comment:
if a saveat point is just below the callback point, does this block it from being saved? Or has savevalues already been called for that?

Reply:
Ah, I didn't think about that case.
```julia
julia> t -= eps(t)
3.1927542840704817

julia> sol = solve(prob,Tsit5(),callback=callback_single,saveat=t); sol.t
7-element Array{Float64,1}:
  0.0
  3.1927542840704817
  3.192754284070482
  6.385508568140963
  9.578262852211415
  9.578262852211445
 15.0
```
Could you give me some pointers about this problem?